### PR TITLE
(backport) asciidoctor: 1.5.7.1 -> 2.0.10

### PR DESCRIPTION
**Pull request by knl**
original message:
There has been a new major version, so it seems good to upgrade.

I had to drop two plugins:
- `asciidoctor-latex` seems dead, and hasn't had any commits since 2015.
- `asciidoctor-bespoke` also hasn't been updated for over a year, and
doesn't seem to be compatible with the new version.

addendum:
`asciidoctor-bespoke` is back

(cherry picked from commit 3d1e0da1a76533149baf96061f900d7c275c5280)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by basvandijk** [comment]
@GrahamcOfBorg build asciidoc

File: pkgs/tools/typesetting/asciidoctor/Gemfile [hunk 1]
```diff
@@ -1,9 +1,7 @@
 source 'https://rubygems.org'
 gem 'asciidoctor'
 gem 'asciidoctor-diagram'
-gem 'asciidoctor-bespoke'
```
Comment (basvandijk) [inline pkgs/tools/typesetting/asciidoctor/Gemfile]: We generally don't like to remove functionality from release branches. Is there a way to bring this back?

Comment (knl) [reply pkgs/tools/typesetting/asciidoctor/Gemfile]: I created one commit on top, that brings back bespoke. @basvandijk PTAL.

**Comment by basvandijk** [comment]
@GrahamcOfBorg build asciidoc

**Pull request merged**